sat/cli/bootsys/mgmt_power.py

```diff
@@ -280,11 +281,18 @@ def do_mgmt_shutdown_power(ssh_client, username, password, excluded_ncns, ncn_sh
             start_shutdown(other_ncns, ssh_client)
             LOGGER.info(f'Waiting up to {ncn_shutdown_timeout} seconds for other NCNs to '
                         f'reach powered off state according to ipmitool: {", ".join(other_ncns)}.')
-            finish_shutdown(other_ncns, username, password,
-                            ncn_shutdown_timeout, ipmi_timeout)
-            LOGGER.info('Shutdown and power off of all other NCNs complete.')
+
+            # Check if the user wants to proceed with hard power off after timeout
+            prompt_message = 'Do you want to forcibly power off the nodes after timeout?'
+            if not hard_power_off(prompt_message):
+                LOGGER.info('User opted not to proceed with hard power off. Exiting.')
+                raise SystemExit(0)
```

Comment:
This code has been added in the wrong place. This prompt will be displayed after the code in `start_shutdown` has issued a `shutdown -h now` command on all the management NCNs but before we have waited for the duration of the `ncn_shutdown_timeout` for them to gracefully shut down and reach a powered off state. The correct place to add such a prompt appears to be in `finish_shutdown`.